franklin/github/views.py

```diff
@@ -257,30 +186,27 @@ def deploy_hook(request):
     return Response(status=status.HTTP_400_BAD_REQUEST)
 
 
-@api_view(['POST'])
+@api_view(['GET', 'POST'])
 @permission_classes((AllowAny, ))
 def get_auth_token(request):
     """
     View to authenticate with github using a client code
-    ---
-
-    type:
-        token:
-            type: string
-            required: true
-            description: oAuth token for the github user
-    parameters:
-        -   name: clientId
-            type: string
-            required: true
-        -   name: redirectUri
-            type: string
-            required: true
-        -   name: code
-            type: string
-            required: true
     """
-
-    logger.info("Received token request from Dashboard")
-
-    return get_access_token(request)
+    if request.method == 'GET':
+        message = {'client_id': os.environ['SOCIAL_AUTH_GITHUB_KEY']}
+        return Response(message, status=status.HTTP_200_OK)
+    elif request.method == 'POST':
+        response = get_access_token(request)
+        if status.is_success(response.status_code):
+            try:
+                access_token = response.json().get('access_token', None)
+                user = do_auth(access_token)
+                serializer = UserSerializer(user)
+                response_data = Response({
+                    'token': access_token,
+                    'user': serializer.data
+                }, status=status.HTTP_200_OK)
+            except KeyError:
+                response_data = Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Why did you change this? 

`response_data = Response({'status': 'Bad request',		
 -                                      'message': 'Authentication could not be\		
 -                                              performed with received data.'},		
 -                                     status=status.HTTP_400_BAD_REQUEST)		
 -        return response_data`